src/main/java/Parser.java

```diff
@@ -0,0 +1,37 @@
+public class Parser {
+
+    public Command parse(String input) throws DukeException {
+        String endCommand = "bye";
+        String listCommand = "list";
+        String doneCommand = "done";
+        String deleteCommand = "delete";
+        String toDoCommand = "todo";
+        String deadlineCommand = "deadline";
+        String eventCommand = "event";
+
+        if (input.equals(listCommand)) {
+            return new ListCommand();
+        } else if (input.equals(endCommand)) {
+            return new ByeCommand();
+        } else {
+            String[] words = input.split(" ", 2);
+            if (words.length < 2) { // some kind of error
+                throw new DukeNotSureException("Huh?? What are you trying to do??");
+            } else if (words[0].equals(doneCommand)) { // the case where a task is done
+                return new DoneCommand(words);
+            } else if (words[0].equals(deleteCommand)) { // the case where a task is deleted
+                return new DeleteCommand(words);
+            } else { // the case where tasks are added
+                if (words[0].equals(toDoCommand)) {
```

Comment:
Perhaps this nested if-else statement can be combined with the parent if-else statement to reduce nesting?

Reply:
Noted, thanks!